README.developer.md

```diff
@@ -14,15 +14,38 @@
   underscore_separated_names.
 - Class names are in CamelCase.
 - Imports should be grouped in the following order:
--- Standard library imports
--- Related third party imports
--- Local application-specific or library-specific imports.
+  - Standard library imports
+  - Related third party imports
+  - Local application-specific or library-specific imports.
 - All indentation is made of the space characters.
   Tabs are evil. In makefiles, however, tabs are obligatory.
   White space between code stanzas are welcome. They help to create breathing
   while reading long code.
   However, splitting stanzas into helper functions could be even better.
 - Prefer single quotes (') whenever possible.
 
-Note: This is a modified version of VDSM conding guidelines:
+## Clean Code
+Do your best to follow the clean code guidelines.
+
+- Name classes using a noun.
+- Name functions/methods using a verb.
+- Make them as small as possible.
+- Should do one thing only and do it well.
+  One thing means one level of abstraction.
+  The names and code should reflect that.
+- Methods/functions should be organized per level of abstraction,
+  where callee sits below their caller.
+- Avoid output-arguments (arguments to output data out of a function/method).
+- Don’t use boolean arguments, use 2 functions/methods instead.
+- Comply to CQS, either:
```

Comment:
Please avoid acronyms without introducing them. Also this is something that is controversial and we do not follow this in the current code and the python stdlib does not as well. I would remove this.